src/Polysemy/ConstraintAbsorber/MonadCatch.hs

```diff
@@ -0,0 +1,115 @@
+{-# LANGUAGE AllowAmbiguousTypes         #-}
+{-# LANGUAGE FlexibleContexts            #-}
+{-# LANGUAGE GADTs                       #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving  #-}
+{-# LANGUAGE RankNTypes                  #-}
+{-# LANGUAGE ScopedTypeVariables         #-}
+{-# LANGUAGE TypeApplications            #-}
+{-# LANGUAGE UndecidableInstances        #-}
+
+module Polysemy.ConstraintAbsorber.MonadCatch
+  (
+    -- * Constraint Absorbers
+    absorbMonadThrow
+  , absorbMonadCatch
+  -- * run helper
+  , runErrorForMonadCatch
+    -- * Re-exports
+  , Exception(..)
+  , SomeException
+  )
+where
+
+import qualified Control.Monad.Catch           as C
+import           Control.Monad.Catch            ( Exception(..)
+                                                , SomeException
+                                                , toException
+                                                )
+import           Polysemy
+import           Polysemy.ConstraintAbsorber
+import qualified Polysemy.Error                as E
+
+
+------------------------------------------------------------------------------
+-- | Like 'E.runError' but applies a given function from 'SomeException'
+-- to some other type, typically something less opaque.
+-- e.g.:
+--  @runErrorForMonadCatch C.displayException@
+-- 
+-- @since 0.7.0.0
+runErrorForMonadCatch
+  :: (C.SomeException -> e)
```

Comment:
This can't be useful, can it? `SomeException` contains an existential and thus has infinite possibilities inside. Maybe this should be a `SomeException -> Maybe e`, which introduces the `Error e` effect into scope?

Reply:
I mean, I use it in the tests since I couldn't use HSpec's ```shouldBe``` with ```SomeException``` since it doesn't have an ```Eq``` instance.  I am assuming the function you give it will almost always invoke ```fromException``` and then handle the ```Maybe``` cases.  But your version doesn't allow the very simple use of ```displayException``` which strikes me as a likely desired use-case.  So I don't know.

Reply:
See https://github.com/adamConnerSax/polysemy-zoo/blob/AddMonadCatch/test/ConstraintAbsorberSpec.hs#L55 
and
https://github.com/adamConnerSax/polysemy-zoo/blob/AddMonadCatch/test/ConstraintAbsorberSpec.hs#L148

Reply:
I'm not super convinced; if you require that function to be total you're just going to force `e` to have a `Nothing` equivalent, eg your `UnknownException`.

Reply:
Except in the displayException case.  Should there be two run functions, one with ```Maybe``` and one for just the display case?  Let me push a version like that and see what you think...

Reply:
Okay, pushed a new version with a text specific run function and one that is explicit with the ```Maybe```. 
I can see dropping these but ```SomeException``` is weird and I'd like to provide some way out of the wilderness for a user who is stuck in the weirdness.